lib/ansible/modules/packaging/os/dnf.py

```diff
@@ -636,6 +636,14 @@ def _mark_package_install(self, pkg_spec, upgrade=False):
         is_newer_version_installed = self._is_newer_version_installed(pkg_spec)
         is_installed = self._is_installed(pkg_spec)
         try:
+            if (pkg_spec in self.base.conf.exclude) or ('*' in self.base.conf.exclude):
```

Comment:
I'm not sure how it works with `dnf`, but with `yum`, the name of the package you're trying to install doesn't necessarily need to be an exact match for what's in `exclude` for the package to be excluded. It's better than nothing, though.

Reply:
@agaffney that's fair, I could try doing some heuristics with `fnmatch`. Thoughts?

Reply:
Is there some built-in method for comparing the `pkg_spec` against an `exclude` entry with a wildcard? There's obviously support for it somewhere in `dnf`, but it may or may not be exposed.

Reply:
I was looking though the [DNF API](https://dnf.readthedocs.io/en/latest/api_queries.html#dnf.query.Query) and I didn't find anything, I also looked through the DNF code itself on how it implements this functionality and it's making API calls that are exposed but are not compatible with older versions of DNF so I'm not sure how I would handle this reliably in a way that would be compatible across DNF v1/v2/v3. I'm still looking though, will update as I find more.